### PR TITLE
ref #128 add TS.MADD

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -485,71 +485,96 @@ void handleCompaction(RedisModuleCtx *ctx, CompactionRule *rule, api_timestamp_t
...
+
+int TSDB_madd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+	   RedisModule_AutoMemory(ctx);
+
+	   if (argc < 4 || (argc-1)%3 != 0) {
+		   return RedisModule_WrongArity(ctx);
+	   }
+
+	   RedisModule_ReplyWithArray(ctx, (argc-1)/3);
+	   for(int i=1; i<argc ; i+=3){
```
Comment (itamarhaber) [inline src/module.c]: This is a problem if there are errors in the loop (e.g. adding to the wrong type of key) - the input must be verified before entering the work loop.

Comment (gkorland) [reply src/module.c]: done

File: src/module.c [hunk 2]
```diff
@@ -485,71 +485,96 @@ void handleCompaction(RedisModuleCtx *ctx, CompactionRule *rule, api_timestamp_t
...
+			RedisModuleString *keyName = argv[i];
+			RedisModuleString *timestampStr = argv[i+1];
+			RedisModuleString *valueStr = argv[i+2];
+			add(ctx, keyName, timestampStr, valueStr, NULL, -1);
```
Comment (danni-m) [inline src/module.c]: @gkorland what happens if some of the adds fail and the other dont??

Comment (gkorland) [reply src/module.c]: you get an array of results and errors see test_partial_madd

**Comment by itamarhaber** [comment]
Also, should be documented.

File: src/module.c [hunk 3]
```diff
@@ -485,71 +485,96 @@ void handleCompaction(RedisModuleCtx *ctx, CompactionRule *rule, api_timestamp_t
...
+
+	    Series *series = NULL;
+
+	    if (argv != NULL && RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
+	        // the key doesn't exist, lets check we have enough information to create one
+	        long long retentionSecs;
+	        long long maxSamplesPerChunk;
+	        size_t labelsCount;
+	        Label *labels;
+	        if (parseCreateArgs(ctx, argv, argc, &retentionSecs, &maxSamplesPerChunk, &labelsCount, &labels) != REDISMODULE_OK) {
```
Comment (danni-m) [inline src/module.c]: @gkorland there's an unexpected outcome for reusing this code. technically you made madd also to create the keys if they don't exists, not a problem really but it opens the door for an interesting use-case.

Comment (gkorland) [reply src/module.c]: madd always calls add with argv == NULL

**Pull request merged**